internal/pkg/conditions/poll.go

```diff
@@ -0,0 +1,86 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Should we look at expanding the retry package for this? Feels similar, but not sure.